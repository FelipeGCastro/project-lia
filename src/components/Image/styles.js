import styled from 'styled-components'
import Skeleton from "react-loading-skeleton";

export const SkeletonStyled = styled(Skeleton)`
  margin: auto;
  display: flex;
  width: 100%;
  object-fit: contain;
  height: 100%;
  max-width: 600;
  max-height: 600;
`
