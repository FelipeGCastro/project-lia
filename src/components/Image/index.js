import React, { useState } from 'react'
import {
  DEFAULT_IMAGE
} from 'src/constants'

import {SkeletonStyled} from './styles'

function Image({src, alt}) {
  const [loaded, setLoaded] = useState(false);

    return (
    <>
    {!loaded ? <SkeletonStyled /> : null}
    <img
      style={loaded ? {} : {display: 'none'}}
      src={src || DEFAULT_IMAGE}
      alt={alt || 'draw'}
      onLoad={() => setLoaded(true)}
    />
  </>
  )
}

export default Image