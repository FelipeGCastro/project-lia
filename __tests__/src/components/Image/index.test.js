// __tests__/CheckboxWithLabel-test.js

import "@testing-library/jest-dom";
// NOTE: jest-dom adds handy assertions to Jest and is recommended, but not required

import React from "react";
import { cleanup, fireEvent, render } from "@testing-library/react";
import Image from "../../../../src/components/Image";
import renderer from "react-test-renderer";

describe("Link changes the class when hovered", () => {
  const component = renderer.create(<Image src="src" alt="alt" />);

  it("Snapshot", () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Snapshot", () => {
    const testInstance = component.root;

    expect(testInstance.findByType(Image).props.alt).toBe("alt");
    expect(testInstance.findByType(Image).props.src).toBe("src");
  });
});
