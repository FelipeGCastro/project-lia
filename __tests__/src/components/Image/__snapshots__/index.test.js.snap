// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered Snapshot 1`] = `
Array [
  <span>
    <span
      className="react-loading-skeleton sc-AxjAm kDlNXv css-1vmnjpn-skeletonStyles-Skeleton"
      style={Object {}}
    >
      â€Œ
    </span>
  </span>,
  <img
    alt="alt"
    onLoad={[Function]}
    src="src"
    style={
      Object {
        "display": "none",
      }
    }
  />,
]
`;
